### PR TITLE
work around cloning repos

### PR DESCRIPTION
**Pull request by CaseyHaralson**
cleaning previous downloads, removing .git dir, and extracting from the cloned folder

### CODE REVIEW

File: src/utils/clone.ts [hunk 1]
```diff
@@ -20,18 +21,29 @@
...
-      } else {
-        reject(new GitlyCloneError('Failed to clone the repository'))
-      }
-    })
-  })
+  const remote = async () => {
+    await removePreviousClone(path)
+
+    return new Promise<string>((resolve, reject) => {
+      const child = spawn('git', ['clone', info.href, path])
```
[the remark this answers is not in the document]
Comment (CaseyHaralson) [reply src/utils/clone.ts:28]: I'm going to try and commit a change to see if I can fix this

[the remark this answers is not in the document]
Comment (CaseyHaralson) [reply src/utils/clone.ts:36]: Well, looks like my test didn't help. I think they are suggesting checking for a valid remote and I thought checking for "upload-pack" would work. 

https://codeql.github.com/codeql-query-help/javascript/js-second-order-command-line-injection/

[the remark this answers is not in the document]
Comment (iwatakeshi) [reply src/utils/clone.ts:38]: Hmm, I wonder if there is a way to ignore that error line...

**Comment by iwatakeshi** [comment]
Superseded by #29

**Pull request closed without merging**